static/js/tab_bar.js

```diff
@@ -90,7 +90,7 @@ exports.exit_search = function () {
         $("#search_query").val(search_string);
     } else {
         // for "searching narrows", we redirect
-        window.location.replace(filter.generate_redirect_url());
+        window.location.href = filter.generate_redirect_url();
```

Comment:
Does it make sense to change these other references?  I guess the other two are basically force-reload type code paths; though I'm not sure that's actually a reason to use replace rather than href.

```
(zulip-py3-venv) tabbott@coset:~/zulip$ git grep window.location[.]replace
static/js/reload.js:    window.location.replace("#reload:" + token);
static/js/setup.js:            window.location.replace(page_params.login_page);
static/js/tab_bar.js:        window.location.replace(filter.generate_redirect_url());
(zulip-py3-venv) tabbott@coset:~/zulip$ git grep window.location[.]href
frontend_tests/node_tests/dispatch.js:    assert_same(window.location.href, "/accounts/deactivated/");
frontend_tests/node_tests/util.js:    window.location.href = "http://zulip.zulipdev.com/";
static/js/blueslip.js:            href: window.location.href,
static/js/click_handlers.js:        window.location.href = $(this).attr('href');
static/js/portico/integrations.js:            window.location = window.location.href;
static/js/server_events_dispatch.js:            window.location.href = "/accounts/deactivated/";
static/js/settings_account.js:                    window.location.href = "/login/";
static/js/util.js:            url = new URL(elt.getAttribute("href"), window.location.href);
static/js/util.js:            url.hash === "" || url.href !== new URL(url.hash, window.location.href).href
```

Reply:
So, the first one doesn't make sense to change, the force refresh is not actually changing the page the user is looking at, so having it append to the history would be annoying.

The second one, however, does make sense. I suppose on of the first things a user might try to do after being logged out is press the back arrow key, in which case they should see that going back redirects them to the log in page again and they have to log in. I've made the change but have NOT manually tested that it's better (not sure how to trigger that block)

Reply:
You can trigger it via logging out in a second browser window and seeing what happens in the first one, I think.

Let's not include this change given that it's not tested; better for it to be in its own PR/commit with appropriate care taken